pyfr/plugins/tavg.py

```diff
@@ -338,4 +346,147 @@ def __call__(self, intg):
 
                     self.tstart_acc = intg.tcurr
 
-                self.tout_last = intg.tcurr
\ No newline at end of file
+                self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(CLIBasePlugin):
+    name = 'tavg'
+    systems = ['*']
+    formulations = ['dual', 'std']
+    cli = [
+        ('merge', 'merge_cli', ('solns', {'nargs': '*', 'help': 'Averages to merge'}), 
+                               ('output', {'help': 'Output file name'})),
+    ]
+
+    def merge_cli(self, args):
+        self.args = args
+        start_time = np.empty((0, 1))
+        end_time = np.empty((0, 1))
+
+        active = self._merge_active_files(args.solns)
+        merged_times = [(r.start, r.end) for r, _, _ in active]
+        std_all = all([s == 'all' for _, _, s in active])
+        avg_time_inv = 1/sum([r.len for r, _, _ in active])
+        window = merge_intervals([r for r, _, _ in active])
+
+        for i, (rng, filename, std_mode) in enumerate(active):
+            f = NativeReader(filename)
+
+            stat = Inifile(f['stats'])
+            fields = stat.get('data', 'fields').split(',')
+
+            # Averager time window for current file
+            dt = rng.len
+
+            if i == 0:
+                uuid = f['mesh_uuid']
+                stat_base = stat
+                fields_base = fields
+                region_base = stat.get('tavg', 'region')
+                
+                cfgs = {k: Inifile(f[k]) for k in f.keys() if k.startswith('config')}
+
+                data_keys = [k for k in f.keys() if k.startswith('tavg_')]
+
+                aidx = []
+                sidx = []
+                self.anames = anames = []
+                self.snames = snames = []
+                for field in fields:
+                    # Truth table to extract collumns later
+                    aidx.append(field.startswith('avg-'))
```

Comment:
Construct these using comprehensions.  So `aidx = [f for f in fields if f.startswith('avg-')]`